src/UncachableEntityPermissionProvider.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Drupal\entity;
+
+use Drupal\Core\Entity\EntityHandlerInterface;
+use Drupal\Core\Entity\EntityPublishedInterface;
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\user\EntityOwnerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides generic entity permissions which are cached per user.
+ *
+ * This includes:
+ *
+ * - administer $entity_type
+ * - access $entity_type overview
+ * - view an ($bundle) $entity_type
+ * - view own ($bundle) $entity_type
+ * - view own unpublished $entity_type
+ * - update (own|any) ($bundle) $entity_type
+ * - delete (own|any) ($bundle) $entity_type
+ * - create $bundle $entity_type
+ *
+ * As this class supports "view own ($bundle) $entity_type" it is just cacheable
```

Comment:
Maybe something like:
"WARNING: This class provides per-user view permissions ("view own"), so the resulting content needs to be cached per-user, which significantly lowers the usefulness of frontend caching. It is therefore recommend to use it only for entity types that are administrative and not user-facing."